src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts

```diff
@@ -110,11 +110,18 @@ export function normalizeClientRequestArgs(
   // and derive request options from it.
   else if (args[0] instanceof URL) {
     url = args[0]
-    logger.info('first argument is a URL:', url)
 
     options = resolveRequestOptions(args, url)
     logger.info('derived request options:', options)
 
+    if (options.path) {
```

Comment:
We seem to have a regression here: since `options` is derived from the `url` a few lines above, what ends up happening is this:

```js
'http://example.com/resource?a=b' // url
{ path: '/resource?a=b' } // options

url.pathname = options.path
```

This will duplicate the query string on the URL since `options.path` rightfully includes query string. 

I will fix this.